logging/alternateServiceLogger.go

```diff
@@ -33,6 +33,18 @@ var zapLevelMap = map[string]zapcore.Level{
 // consoleLog if set to false outputs in a json like format (Though can have duplicate keys which downstream processors may handle in undefined ways).
 //   Formats the log in a more traditional one line fashion with a leading timestamp and log level.
 func NewZapSugaredServiceLogger(out string, serviceName string, level string, initialFields map[string]interface{}, consoleLog bool) AltServiceLogger {
+	return NewZapSugaredServiceLoggerWithSkip(out, serviceName, level, initialFields, consoleLog, 1)
+}
+
+// NewZapSugaredServiceLoggerWithSkip returns a logger with designated logging levels for a particular service.
+// out is the location for logs to be output such as "stdout"
+// service is the value for the "service" key.
+// level is the minimum logging level that will be output
+// initialFields is a map of key value pairs that will be logged with all log message produced by this logger.
+// consoleLog if set to false outputs in a json like format (Though can have duplicate keys which downstream processors may handle in undefined ways).
+//   Formats the log in a more traditional one line fashion with a leading timestamp and log level.
+// Skip level is used for configuring the caller line number. Services often want 1
+func NewZapSugaredServiceLoggerWithSkip(out string, serviceName string, level string, initialFields map[string]interface{}, consoleLog bool, skipLevel int) AltServiceLogger {
```

Comment:
do we need another method or because this is only used in one place which is going to be updated based off this pr, can we collect all these params into a struct and add this skip level as a field?

Reply:
It is not yet picked up anywhere else so we could rearrange this into a struct I am not opposed to that. Will do

Reply:
I just wish there could be default values because you basically never want it to be zero